TicTacToe.sol

```diff
@@ -84,76 +59,61 @@ contract TicTacToe {
 //       sources = transactions[transactionId].sources; // In case you have array of transactions
 //     }
 
-    function getBoard(uint256 roomNumber) public view returns (int8[9] memory board) {
-        board = rooms[roomNumber].board;
+    function getBoard(uint256 roomNumber) public view returns (int8[9] memory) {
+        return rooms[roomNumber].board;
     }
 
     function hasWinner(uint8 roomIndex) public view returns (address) {
         for (uint32 i = 0; i < winningCases.length; i++) {
-            WinningCase memory currentCase = winningCases[i];
-            uint firstIndex = currentCase.firstIndex;
-            uint secondIndex = currentCase.secondIndex;
-            uint thirdIndex = currentCase.thirdIndex;
+            uint8[3] memory currentCase = winningCases[i];
+            uint firstIndex = currentCase[0];
+            uint secondIndex = currentCase[1];
+            uint thirdIndex = currentCase[2];
+
+            Room memory room = rooms[roomIndex];
+
             if (
-                rooms[roomIndex].board[firstIndex] ==
-                rooms[roomIndex].board[secondIndex] &&
-                rooms[roomIndex].board[secondIndex] ==
-                rooms[roomIndex].board[thirdIndex] &&
-                rooms[roomIndex].board[firstIndex] != EMPTY_MOVE
+                room.board[firstIndex] != EMPTY_MOVE &&
```

Comment:
Check if there's an `EMPTY_MOVE` first can save gas by not going through the other checks